### PR TITLE
feat(chat): add delete shared entities (Issue #709)

### PR DESCRIPTION
**Pull request by mikitabut**

### CODE REVIEW

**Comment by mikitabut** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-774.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by mikitabut** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-774.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by mikitabut** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-774.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by mikitabut** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-774.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by mikitabut** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-774.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/store/share/share.epics.ts [hunk 1]
```diff
@@ -476,49 +473,242 @@ const getSharedListingSuccessEpic: AppEpic = (action$, state$) =>
...
+          ConversationsActions.updateConversationSuccess({
+            id: payload.resourceId,
+            conversation: {
+              isShared: false,
+            },
+          }),
+        );
+      }
+      if (
+        payload.nodeType === 'FOLDER' &&
```
Comment (Alexander-Kezik) [inline apps/chat/src/store/share/share.epics.ts]:
```suggestion
        payload.nodeType === BackendDataNodeType.FOLDER &&
```

File: apps/chat/src/store/share/share.epics.ts [hunk 2]
```diff
@@ -476,49 +473,242 @@ const getSharedListingSuccessEpic: AppEpic = (action$, state$) =>
...
+          ConversationsActions.updateFolder({
+            folderId: payload.resourceId,
+            values: {
+              isShared: false,
+            },
+          }),
+        );
+      }
+      if (
+        payload.nodeType === 'ITEM' &&
```
Comment (Alexander-Kezik) [inline apps/chat/src/store/share/share.epics.ts]:
```suggestion
        payload.nodeType === BackendDataNodeType.ITEM &&
```

File: apps/chat/src/store/share/share.epics.ts [hunk 3]
```diff
@@ -476,49 +473,242 @@ const getSharedListingSuccessEpic: AppEpic = (action$, state$) =>
...
+          PromptsActions.updatePromptSuccess({
+            id: payload.resourceId,
+            prompt: {
+              isShared: false,
+            },
+          }),
+        );
+      }
+      if (
+        payload.nodeType === 'FOLDER' &&
```
Comment (Alexander-Kezik) [inline apps/chat/src/store/share/share.epics.ts]:
```suggestion
        payload.nodeType === BackendDataNodeType.FOLDER &&
```
Please, also check all the places below

**Review by Alexander-Kezik — APPROVED** [review]

**Pull request merged**